### ISSUE TITLE
deno fmt eliding name type in mapped types and inserting semicolons

### ISSUE THREAD

**Issue by sno2** [denoland/deno#9561]
I have a file and when I try to format it with `deno fmt` it has the following results:

```ts
//////////////////////////////////////////////////////////////////////////////////////////////
////                                                                                     /////
////                SECTION 1: Global Setup Code                                         /////
////                                                                                     /////
//////////////////////////////////////////////////////////////////////////////////////////////

export enum NodeType {
  Element,
  /** @deprecated */
  Attribute,
  Text,
  CDataSection,
  /** @deprecated */
  EntityReference,
  /** @deprecated */
  Entity,
  ProcessingInstruction,
  Comment,
  Document,
  DocumentType,
  DocumentFragment,
  /** @deprecated */
  Notation,
}

//////////////////////////////////////////////////////////////////////////////////////////////
////                                                                                     /////
////      SECTION 2: Removing as clause in computed properties of a typedef              /////
////                                                                                     /////
//////////////////////////////////////////////////////////////////////////////////////////////

// before formatting
export interface Node {
  readonly nodeType: NodeType | keyof { [K in NodeType as `${typeof NodeType[K]}`]: undefined };
}

// after
export interface Node {
  readonly nodeType: NodeType | keyof { [K in NodeType]: undefined };
}

//////////////////////////////////////////////////////////////////////////////////////////////
////                                                                                     /////
////      SECTION 3: Adding Semi-colon even when ignoring that line                      /////
////                                                                                     /////
//////////////////////////////////////////////////////////////////////////////////////////////

// before formatting
export interface Node {
  // deno-fmt-ignore
  readonly nodeType: NodeType | keyof { [K in NodeType as `${typeof NodeType[K]}`]: undefined };
}

// after
export interface Node {
  // deno-fmt-ignore
  readonly nodeType: NodeType | keyof { [K in NodeType as `${typeof NodeType[K]}`]: undefined };; // <- note the extra semi-colon
}
```

I am using the latest version of Deno (1.7.5).

**Comment by sno2** [comment]
Similar to #9477

**Comment by kitsonk** [comment]
In the first case it looks like `fmt` isn't handling the string literal types introduced in TS 4.1.

**Comment by sno2** [comment]
> In the first case it looks like `fmt` isn't handling the string literal types introduced in TS 4.1.

Yeah

**Comment by sno2** [comment]
I ran into the issue when I made [this stackoverflow answer](https://stackoverflow.com/a/66288492/10873797) which allowed you to get the keys and values of an enum in types using the as clause.  If anyone wants a simpler example for when the bug would occur.

**Comment by dsherret** [comment]
We should give this and #9477 more descriptive titles. They could both match anything to do with deno fmt.

Opened https://github.com/dprint/dprint-plugin-typescript/issues/111

**Comment by kitsonk** [comment]
Yeah, sorry, it was the weekend and I wasn't near my laptop to open the dprint issues.

**Comment by dsherret** [comment]
@kitsonk that's no trouble!

By the way, in this case it wasn't template literal types not being formatted, but the name type in mapped types wasn't being included in the output.

**Comment by dsherret** [comment]
This issue should be fixed now since main has 0.41.0 of dprint-plugin-typescript.
